### DISCUSSION TITLE
Turn on device for a limited time

### DISCUSSION THREAD

**Discussion by Toby-Mills** [andrei-tatar/node-red-contrib-smartnora#109] [Q&A] [answered]
Do any of the supported device types allow the specification of a period of time?
I'd like to be able to specify that an irrigation station is turned on for 30 minutes, or that a light is turned on for 5 mins.

**Comment by andrei-tatar** [comment, marked as answer]
This should be supported by Google Home by default (implicitly by Smart NORA)
More details here: https://developers.google.com/assistant/smarthome/develop/scheduling

**Reply by Toby-Mills** [reply]
Ah! I didn't know you could do that! I see that it sends through an 'Off' payload after the specified amount of time.
It's not quite what I had hoped for, but I can probably work with it.
Thanks :)

**Comment by Syridian** [comment]
Hi Andrei,

Google limits to certain devices and won't allow "switches" and "outlets" to be scheduled.  Would it be possible to either have a special node that allows for configuration items to be scheduled that gets around this, or have a tick box in the node that allows us to turn off that "safety feature"?

My particular use case is that I have motion sensors for my lighting and I use node-red to ignore or respond to these by asking google to turn on or off the sensor, however, I can't schedule these as the only way to get an on/off response from Smart NORA is the switch or outlet.  Or have I missed a node that will allow this?

**Reply by Toby-Mills** [reply]
@Syridian I see Smart Nora has support for 'Sensor'. Have you tried to see if Google will allow that to be scheduled?

**Reply by andrei-tatar** [reply]
> Google limits to certain devices and won't allow "switches" and "outlets" to be scheduled. Would it be possible to either have a special node that allows for configuration items to be scheduled that gets around this, or have a tick box in the node that allows us to turn off that "safety feature"?

Unfortunately there's no way of telling Google not to limit the device :(

> My particular use case is that I have motion sensors for my lighting and I use node-red to ignore or respond to these by asking google to turn on or off the sensor, however, I can't schedule these as the only way to get an on/off response from Smart NORA is the switch or outlet. Or have I missed a node that will allow this?

So you want to have the motion sensors on/off based on a schedule but also use voice actions to turn them on/off immediately? I would keep them as switches in NORA and handle the scheduling part from node-red.

I think it's better to keep scheduling separate from NORA since it's a different problem. There are lots of nodes that tackle this: https://flows.nodered.org/collection/c8156f6276976bfb518d1e60442e01e2 

I have a similar use case where I made a dog feeder. It starts automatically once a day at 8:30 but I can also turn it on by voice to give a snack to my dogo. I have a simple inject node that sends a "turn on" command to the feeder every day at 8:30
![image](https://user-images.githubusercontent.com/6473183/145684717-e71b598c-f99a-4593-911a-077b8241db51.png)

**Reply by Syridian** [reply]
Hi @Toby-Mills, 

The sensor node doesn't seem to have an output that I could use for turning something on or off.  Maybe I need to investigate it further.

Hi @andrei-tatar, 

I actually want the motion sensors to be turned on/off on an ad-hoc basis.  The majority of the time they are on, but it would be nice to have the ability to ask Google to turn off the motion sensors for a certain period whilst I'm doing stuff in a particular room.  Working at a desk for long periods where I'm not moving enough for the 5 minute period, having the ability to say "Hey Google, turn off the motion sensor for 2 hours" would be a huge benefit.  Scheduling this in node-red is just impractical.  Maybe @Toby-Mills suggestion of using the sensor could be a neat solution if it could also output an on/off msg?

Though I must admit, there's a number of new nodes in Smart NORA since I first started playing so there might be something I'm missing which will allow that.

**Reply by andrei-tatar** [reply]
Probably the Timer trait (https://developers.google.com/assistant/smarthome/traits/timer) offers the closest thing you want to do though the voice command would probably need to be something like "Hey Google, start a timer for 2 hours for the sensors" or something like that. Anyways, the timer trait is not yet supported.
